auth/role_info.hh

```diff
@@ -36,4 +45,64 @@ struct role_config_update final {
     std::optional<bool> can_login{};
 };
 
+struct role_info : public data_dictionary::keyspace_element {
+    sstring role_name;
```

Comment:
A role isn't a keyspace element. Tables and functions are all nested under keyspaces, but roles are top-level objects (they're keyspace siblings, not children).

Reply:
yes, the abstraction leaks, I am not sure if all those abstractions are very useful, they hide implementation, but what besides that? there is only a single simple usage of it - describe.

I imagined the whole describe auth implemenation rather as a single function using existing auth getters to fetch everything and then fmt::format(...) to a string in a way that executing the string recreates stuff.

Reply:
> yes, the abstraction leaks, I am not sure if all those abstractions are very useful, they hide implementation, but what besides that? there is only a single simple usage of it - describe.

If it's only used by describe, then perhaps we can just change the name to `schema_element` or something similar and e.g. adjust `keyspace_name()` to return a `std::optional<sstring>` to represent the null.

> I imagined the whole describe auth implemenation rather as a single function using existing auth getters to fetch everything and then fmt::format(...) to a string in a way that executing the string recreates stuff.

Yeah, it looks like a keyspace_element / schema_element could be just a simple struct - the fields could be converted to getters.

@Jadw1 do you remember why the `keyspace_element` interface was introduced in the first place? Is there a reason why it couldn't just be a struct with public fields?

Reply:
I did it to unified way to getting description of keyspace/table/index/view/... but at that point I didn't know we were going to extend describe to auth and service levels. 
Can't we just rename `keyspace_element` class to include broader classes of elements, keep existing implementations of keyspace/table/... descriptions and return objects implementing renamed `keyspace_element` for roles and service levels?

Reply:
I took a look at how `keyspace_element` is used in the codebase right now, and it looks like it is used for two purposes:

- Its methods are only used in the `cql3::description`'s constructor,
- It is also used to be able to put such keyspace elements into `std::vector<shared_ptr<const keyspace_element>>` and then pass them to `generate_descriptions.

I suggest doing the following steps to get rid of the `keyspace_element` interface:

1. Change the existing code in `describe_statement.cc` to generate descriptions right away and collect them to a vector and not the `keyspace_element`s. Then, `generate_descriptions` becomes obsolete with the exception of one thing - it sorts the items by `element_name()`; we can just have `sort_descriptions` function which sorts them by element name.
3. To all existing implementors of `keyspace_element`, add a new method `generate_description` which will return a description. There is no need for it to be a virtual method. Get rid of the current constructors of `cql3::description` and replace them with calls to the new method.
3. At this point, it should be able to just remove the `keyspace_element` interface and the methods implemented for it.